src/library/tags/tag.h

```diff
@@ -0,0 +1,310 @@
+#pragma once
+
+#include <QJsonValue>
+#include <QVector>
+
+#include "util/assert.h"
+#include "util/macros.h"
+#include "util/math.h"
+#include "util/optional.h"
+
+namespace mixxx {
+
+namespace library {
+
+namespace tags {
+
+/// A category for tags.
```

Comment:
Please clarify that this is a Mixxx feature and not to be conflated with file metadata tags.

Reply:
I have added the explanation in the docs for `Tag`. Might be moved somewhere else if we find a better place.